homotopy-web/src/buffers.rs

```diff
@@ -573,6 +577,104 @@ impl Bufferer for CylinderWireBufferer {
     }
 }
 
+struct DebugBufferer;
+
+struct DebugBuffererState {
+    elements: Vec<u16>,
+    verts: IdxVec<u16, Vec3>,
+}
+
+pub struct DebugArrayData {
+    pub element_buffer: ElementBuffer,
+    pub vert_buffer: Buffer<Vec3>,
+}
+
+impl BuffererState for DebugBuffererState {
+    type VertexData = Vec3;
+
+    fn alloc() -> Self {
+        Self {
+            elements: Vec::with_capacity(VAO_LIMIT),
+            verts: IdxVec::with_capacity(VAO_LIMIT),
+        }
+    }
+
+    fn push_vert(&mut self, v: u16, data: Self::VertexData) {
+        let i = self.verts.push(data);
+
+        debug_assert_eq!(i, v);
+    }
+}
+
+impl Bufferer for DebugBufferer {
+    type Vertex = Vert;
+    type Output = DebugArrayData;
+    type State = DebugBuffererState;
+    type Key = ();
+
+    fn new(_geom: &SimplicialGeometry) -> Self {
+        Self
+    }
+
+    fn buffer(ctx: &mut BufferingCtx<Self>) -> Result<()> {
+        for (n, (tri, _)) in ctx.geom.areas.values().copied().enumerate() {
```

Comment:
If you use a different geometry which contains all the points/lines (see my earlier comment about), you wouldn't need this horrible hack.

Reply:
The reason this is here is because in what gets drawn, each edge is pushed together (to make a gap for the node label). This means that there's no longer a one-to-one correspondence between logical points in the geometry and points that get drawn by the renderer, rather each logical point spawns a cloud of rendered points, one for each edge touching it.

Reply:
Yeah I get that. What I meant is you're using the triangles to construct the points/lines (which is ugly) and you won't need to do once you do what I said in my comment above.

Reply:
Oh you were talking about the areas part rather than the enumerate part. Yeah, I see that. I think keeping these indices is probably a hack too but I'm not too sure how to fix it.

Reply:
Also the correct way to do this is to "inflate" the geometry before you pass it to the bufferer, like Nathan did for cylinders and spheres.